droidlet/dashboard/web/src/components/Memory2D.js

```diff
@@ -113,33 +119,209 @@ class Memory2D extends React.Component {
     });
   };
   handleObjClick = (obj_type, map_pos, data) => {
-    let { popup_coords } = this.state;
-
-    this.setState({
-      table_visible: true,
-      table_coords: map_pos,
-      table_data: data,
-    });
-
-    // close other tabular elements when switching to new map_pos
-    if (map_pos[0] !== popup_coords[0] || map_pos[1] !== popup_coords[1]) {
-      this.setState({ popup_visible: false });
+    let { grouping_mode, grouped_objects, focused_point_coords } = this.state;
+
+    if (!grouping_mode) {
+      // if not in grouping mode, open MemoryMapTable
+      this.setState({
+        table_data: data,
+        table_visible: true,
+        focused_point_coords: map_pos,
+      });
+      // close other tabular elements when switching to new obj/cluster
+      if (
+        map_pos[0] !== focused_point_coords[0] ||
+        map_pos[1] !== focused_point_coords[1]
+      ) {
+        this.setState({ popup_visible: false });
+      }
+    } else {
+      // otherwise if in grouping_mode..
+      if (data.memid in grouped_objects) {
+        // ..unselect object
+        let { [data.memid]: _, ...rest } = grouped_objects;
+        this.setState({ grouped_objects: rest });
+      } else {
+        // ..select object
+        this.setState({
+          grouped_objects: {
+            ...grouped_objects,
+            [data.memid]: data,
+          },
+        });
+      }
     }
   };
   handlePopupClick = (map_pos, data) => {
-    let { table_coords } = this.state;
+    let { focused_point_coords } = this.state;
+
+    // TODO: select/unselect all objects in cluster when using Cmd + Click
```

Comment:
cmd+click is a feature we don't plan to implement, so remove this "TODO", because that implies technical debt, not "feature we don't want."